Fixed issue with existing route config

gendronb opened the pull request:
Most recent release(s) of this plugin breaks my existing Fastify projects that are using per-route configuration (used for another plugin I made)...

I'm submitting this quick fix (and additional test) to solve this issue.

mcollina commented on index.js:
```diff
@@ -55,15 +55,17 @@ function rateLimitPlugin (fastify, settings, next) {
+          // if the current endpoint have a custom rateLimit configuration ...
+          buildRouteRate(current, makeParams(routeOptions.config.rateLimit), routeOptions)
+        } else if (routeOptions.config.rateLimit === false) {
+          // don't apply any rate-limit
```
I don't think this condition could actually be triggered, because `routeOptions.config.rateLimit` is already going to be truthy. Would you mind removing it?

gendronb replied:
Hi @mcollina you're right, in fact, my fix solves some issues, but not all. The problem is that all tests are made without the assumption of existing per-route configuration for other plugins.

For example, if I add a route config to the other tests, most of them will fail even with my fix.

I will submit an updated PR later today that solves all use-cases...

mcollina commented:
Good spot, thanks!

gendronb commented:
Hi again @mcollina...

I have simplified the config check statement (to a single line), now works with any pre-existing per-route config.

I also updated all per-route tests to include a default per-route config (to insure we don't break or conflict with other plugin configuration)...

mcollina commented on index.js:
```diff
@@ -65,6 +65,7 @@ function rateLimitPlugin (fastify, settings, next) {
       } else {
         throw new Error('Unknown value for route rate-limit configuration')
       }
+      // }
```
Can you remove this line?

gendronb replied:
Done...

mcollina commented:
Thanks!
